File: modules/AcroStack.Chat/Dtos/EditMessageInput.cs
using System.ComponentModel.DataAnnotations;

namespace AcroStack.Chat;

/// <summary>Input for editing an existing chat message's text.</summary>
public class EditMessageInput
{
    /// <summary>
    /// 新的消息文本。限制最长 4000 字符，与 ChatMessage.Text 的数据库列宽
    /// （见 ChatEfCoreDbContextExtensions.MaxMessageTextLength）一致，
    /// 避免超长文本穿透到数据库层才抛异常。
    /// </summary>
    [StringLength(4000)]
    public string Text { get; set; } = string.Empty;
}

File: react/src/api/models/acroStack/chat/EditMessageInput.ts
/* oxlint-disable */

/**
 * @description Input for editing an existing chat message\'s text.
 * @type object
 */
export type AcroStackChatEditMessageInput = {
  /**
   * @description 新的消息文本。限制最长 4000 字符，与 ChatMessage.Text 的数据库列宽\r\n（见 ChatEfCoreDbContextExtensions.MaxMessageTextLength）一致，\r\n避免超长文本穿透到数据库层才抛异常。
   * @minLength 0
   * @maxLength 4000
   * @type string | undefined
   */
  text?: string | null;
};
